# Обзор данных

Набор данных представляет собой квартальную отчётность по всем объявлениям в городе Ашвилл (Asheville, North Carolina, United States) на момент 22 сентября 2025 года.

Цель данного ноутбука - зафиксировать общий состав набора данных, его размер, структуру признаков и первые ограничения, которые видны ещё до этапа технической очистки.

## Загрузка библиотек и установка настроек

In [1]:
import pandas as pd

pd.set_option("display.float_format", "{:.4f}".format)


## Загрузка набора данных

In [2]:
listings = pd.read_csv("../data/raw/listings.csv")


## Базовые характеристики набора данных

### Количество объектов и признаков

In [3]:
listings.shape


(2852, 18)

У нас имеется *2852* объектов для исследования и *18* признаков в каждом.

### Схема набора данных

In [4]:
listings.info()


<class 'pandas.DataFrame'>
RangeIndex: 2852 entries, 0 to 2851
Data columns (total 18 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              2852 non-null   int64  
 1   name                            2852 non-null   str    
 2   host_id                         2852 non-null   int64  
 3   host_name                       2849 non-null   str    
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   2852 non-null   int64  
 6   latitude                        2852 non-null   float64
 7   longitude                       2852 non-null   float64
 8   room_type                       2852 non-null   str    
 9   price                           2537 non-null   float64
 10  minimum_nights                  2852 non-null   int64  
 11  number_of_reviews               2852 non-null   int64  
 12  last_review                     2609 non-null

**Идентификаторы и общие сведения**  
**id** - уникальный идентификатор объявления  
**name** - название объявления  
**host_id** - уникальный идентификатор хоста  
**host_name** - имя хоста  

**География объекта**  
**neighbourhood_group** - более крупная административная единица, чем район. В данном наборе данных полностью отсутствует  
**neighbourhood** - код района  
**latitude** - географическая широта  
**longitude** - географическая долгота  

**Характеристики предложения**  
**room_type** - категория объекта: `Entire home/apt`, `Private room`, `Shared room`, `Hotel room`  
**price** - цена за ночь  
**minimum_nights** - минимальное число ночей для аренды  
**availability_365** - количество дней в году, когда объект доступен для бронирования  
**license** - лицензия на объект. В данном наборе данных полностью отсутствует  

**Активность и спрос**  
**number_of_reviews** - общее число отзывов на объект  
**last_review** - дата последнего отзыва  
**reviews_per_month** - среднее число отзывов в месяц  
**calculated_host_listings_count** - количество активных объектов у данного хоста  
**number_of_reviews_ltm** - количество отзывов на объект за последние 12 месяцев (`ltm` = `last twelve months`)

### Первичная оценка пропусков

In [14]:
listings.isna().mean().sort_values(ascending=False)


license                          1.0000
neighbourhood_group              1.0000
price                            0.1104
reviews_per_month                0.0852
last_review                      0.0852
host_name                        0.0011
name                             0.0000
id                               0.0000
neighbourhood                    0.0000
host_id                          0.0000
room_type                        0.0000
longitude                        0.0000
latitude                         0.0000
number_of_reviews                0.0000
minimum_nights                   0.0000
calculated_host_listings_count   0.0000
availability_365                 0.0000
number_of_reviews_ltm            0.0000
dtype: float64

Заметим, что столбцы `license` и `neighbourhood_group` отсутствуют полностью, а признаки `price`, `last_review` и `reviews_per_month` заполнены не для всех объявлений.

Пропуски в `last_review` и `reviews_per_month` могут быть связаны с объявлениями, у которых ещё не было отзывов.    
Пропуски в `price`, вероятно, связаны с объявлениями, в которых цены являются динамическими или обсуждаемыми.

### Вид объектов

In [6]:
listings.head()


,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license
0,155305,Cottage! BonPaul + Sharky's Hostel,746673,BonPaul,NaN,28806,35.5786,-82.5958,Entire home/apt,96.0000,1,463,2025-09-01,2.6900,8,70,23,NaN
1,197263,Tranquil Room & Private Bath,961396,Timothy,NaN,28806,35.5774,-82.6380,Private room,44.0000,2,90,2025-09-01,0.5600,2,88,3,NaN
2,209068,Terrace Cottage,1029919,Kevin,NaN,28804,35.6176,-82.5518,Entire home/apt,90.0000,30,67,2025-05-03,0.3900,1,268,2,NaN
3,246315,Asheville Dreamer's Cabin,1292070,Annie,NaN,28805,35.5962,-82.5063,Private room,62.0000,7,53,2019-10-30,0.3100,3,70,0,NaN
4,314540,Asheville Urban Farmhouse Entire Home 4.6 mi t...,381660,Tom,NaN,28806,35.5856,-82.6273,Entire home/apt,221.0000,1,36,2025-06-23,0.2200,1,180,7,NaN


Замечания:  
Структура таблицы соответствует описанию признаков.  
`neighbourhood` является ZIP-кодом района.  
`last_review` пока хранится как строковое значение.     
Объявления сильно различаются по цене, минимальному сроку аренды и накопленному числу отзывов.

## Промежуточные выводы

На этапе первичного обзора можно сделать несколько выводов.

1. Набор данных содержит достаточный объём наблюдений для дальнейшего исследовательского анализа.
2. Два столбца полностью пусты и не будут использоваться в дальнейшем анализе.
3. Пропуски в `price`, `last_review` и `reviews_per_month` требуют дополнительной обработки, а также корректная интерпретация полей `neighbourhood` как categorical и `last_review` как datetime типов.    